python/sdk_e2e/test_track_1/test_pnp_properties.py

```diff
@@ -20,95 +22,309 @@ def client_kwargs(pnp_model_id):
     return {"model_id": pnp_model_id}
 
 
+@pytest.fixture(
+    params=[
+        pytest.param("int", id="integer value"),
+        pytest.param("bool", id="boolean value"),
+        pytest.param("str", id="string value"),
+        pytest.param("dict", id="complex value"),
+        pytest.param("guid", id="guid value"),
+    ]
+)
+def random_property_value(request, random_dict_factory, random_string_factory):
+    param = request.param
+    if param == "int":
+        return random.randint(-65535, 65535)
+    elif param == "bool":
+        return random.random() > 0.5
+    elif param == "str":
+        return random_string_factory()
+    elif param == "dict":
+        return random_dict_factory()
+    elif param == "guid":
+        return str(uuid.uuid4())
+
+
+def make_desired_property_patch(component_name, property_name, property_value):
+    logger.info("Setting {} to {}".format(property_name, property_value))
+    if component_name:
+        return {
+            Fields.THIEF: {
+                Fields.CMD: Commands.UPDATE_PNP_PROPERTIES,
+                Fields.PNP_PROPERTIES_UPDATE_PATCH: [
+                    {
+                        "op": "add",
+                        "path": "/{}".format(component_name),
+                        "value": {property_name: property_value, "$metadata": {}},
+                    }
+                ],
+            }
+        }
+    else:
+        return {
+            Fields.THIEF: {
+                Fields.CMD: Commands.UPDATE_PNP_PROPERTIES,
+                Fields.PNP_PROPERTIES_UPDATE_PATCH: [
+                    {"op": "add", "path": "/{}".format(property_name), "value": property_value}
+                ],
+            }
+        }
+
+
 @pytest.mark.pnp
+@pytest.mark.parametrize(
+    "is_component_property",
+    [pytest.param(True, id="component property"), pytest.param(False, id="root property")],
+)
 @pytest.mark.describe("Device Client PNP properties")
 class TestPnpSetProperties(object):
     @pytest.mark.it(
-        "Can set a root reported property value and retrieve it via the service get_digital_twin function"
+        "Can set a reported property value and retrieve it via the service get_digital_twin function"
```

Comment:
Are "reported property" and "desired property" terms we should be using in the context of PnP? Here and elsewhere in the code.